dali/operators/image/remap/rotate_params.h

```diff
@@ -93,11 +91,11 @@ inline TensorShape<3> RotatedCanvasSize(TensorShape<3> input_shape, vec3 axis, d
   // source dimension - this helps reduce the blur in central area, especially for
   // small angles.
   for (int i = 0; i < 3; i++) {
-    if (out_size[i] % 2 != in_size[dominant_src_axis[i]] % 2)
-      out_size[i]++;
+    // kernels::vec2shape reverses the extents, store them in that order in parity vector
+    parity[2 - i] = in_size[dominant_src_axis[i]] % 2;
```

Comment:
what's wrong with storing it naturally and returning `vec2shape(parity)`?

Reply:
Because it has `+` operator defined:)

Reply:
Fair point :) But now that you've mentioned it, it also has a `max`. How about switching the order to x, y, z and returning two vectors? You could calculate the size on ivec2/3 and convert to TensorShape at the very end.

Reply:
Great, thanks for pointing to it.